src/script/builder/tir/prim_func_frame.cc

```diff
@@ -62,16 +68,88 @@ tvm::tir::Var Arg(String name, tvm::tir::Var var) {
   return var;
 }
 
-tvm::tir::Buffer Arg(String name, tvm::tir::Buffer buffer) {
+tvm::tir::Var Arg(String name, tvm::tir::Buffer buffer) {
```

Comment:
Why are we doing this change though? I thought it's used for cases like:

```python
def func(
  A: T.Buffer(...),  #  <=== used here
) -> None: ...
```

which the parser interprets as:

```python
A = T.arg("A", T.Buffer(...))
```

Did I miss anything?